Civi/Sqltasks/Actions/RunSQLTask.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Civi\Sqltasks\Actions;
+
+use \Civi\ActionProvider\Action\AbstractAction;
+use \Civi\ActionProvider\Exception\ExecutionException;
+use \Civi\ActionProvider\Parameter\ParameterBagInterface;
+use \Civi\ActionProvider\Parameter\SpecificationBag;
+use \Civi\ActionProvider\Parameter\Specification;
+
+use CRM_Sqltasks_ExtensionUtil as E;
+
+class RunSQLTask extends AbstractAction {
+
+  /**
+   * Run the action
+   *
+   * @param ParameterInterface $parameters
+   * @param ParameterBagInterface $output
+   * @return void
+   */
+  protected function doAction(ParameterBagInterface $parameters, ParameterBagInterface $output) {
+    // Configuration
+    $task_id = $this->configuration->getParameter('task_id');
+    $log_to_file = $this->configuration->getParameter('log_to_file');
+    $return_parameter = $this->configuration->getParameter('return_parameter');
+
+    // Parameters
+    $input_values = $parameters->getParameter('input_values');
```

Comment:
Is there a way to get some sort of key-value data structure here, rather than a "flat" array with input values?

I configured my action with 4 inputs, which seems to result in something like `["206","Patrick","Figel","patrick@figel.email"]`; ideally, it would be something like `["id":"206"]`, otherwise it's hard to build robust SQL Tasks (they'd be dependent on parameter order and what not).